vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/rest/HtsgetVcfController.java

```diff
@@ -111,6 +110,9 @@ public ResponseEntity getHtsgetUrls(
         }
         if (end == null) {
             end = controller.getCoordinateOfLastVariant(referenceName);
+        } else {
+            //end is exclusive
+            end--;
```

Comment:
you did this after `validateParameters`, which returns error if `end <= start`. Using inclusive ranges the behaviour should be:
- start=100, end=99 should give error.
- start=100, end=100 is ok: a range of a single base.
- start=100, end=101+ is ok, bigger range.

this is expressed as `if ( ...not null... && end < start) { error }` *with inclusive ranges* (so actually there was a bug). If all the code assumes inclusive ranges, I would transform the exclusive end into inclusive as soon as possible, even before `validateParameters`.